cookbooks/aws-parallelcluster-slurm/files/default/head_node_slurm/slurm/pcluster_slurm_config_generator.py

```diff
@@ -228,7 +240,7 @@ def _gpu_type(instance_type):
 
 def _vcpus(compute_resource) -> int:
     """Get the number of vcpus for the instance according to disable_hyperthreading and instance features."""
-    instance_type = compute_resource["InstanceType"]
+    instance_type = _instance_type_from_compute_resource(compute_resource)
```

Comment:
#TODO use this also in `def _realmemory` so to account single instance type in InstanceTypeList